### ISSUE TITLE
Button outline on focus: hide or keep?

### ISSUE THREAD

**Issue by andfinally** [Automattic/wp-calypso#21803]
Is there a policy about the outline Chrome adds to a restyled button when it has focus? It seems to be generally overridden in buttons in Calypso. It can detract from the clean look of the UI, but apparently it has [accessibility benefits](https://stackoverflow.com/a/25298082/678834) for people who navigate with a keyboard.

#### Steps to reproduce

Hard to reproduce, as the buttons concerned are still in a [PR](https://github.com/Automattic/wp-calypso/pull/21712). But the outline appears in Chrome after you've clicked on one of the arrow buttons shown in the screenshot, i.e. when the button has focus.

#### What I expected

This is not unexpected, it's commonly seen on sites which do not remove the outline in their CSS.

#### Browser / OS version

macOS High Sierra, Chrome 63

#### Screenshot / Video

![image](https://user-images.githubusercontent.com/1647564/35353362-db275ede-013e-11e8-90c2-62dbedbd485c.png)

#### Context / Source

Manual testing.

**Comment by simison** [comment]
Calypso has [`accessible-focus`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/accessible-focus/README.md) module. So it seems that if you managed to get that outline visible _using a mouse_, there's something off. It should be visible when navigating _using a keyboard_.

I like to refer to awesome @a11yproject whenever in doubt with a11y: 
https://a11yproject.com/posts/never-remove-css-outlines/

TL;DR:
>Use of the rule `:focus { outline: none; }` to remove an outline results in the link or control being focusable but with no visible indication of focus for keyboard users.

**Comment by shaunandrews** [comment]
> Is there a policy about the outline Chrome adds to a restyled button when it has focus? It seems to be generally overridden in buttons in Calypso.

The focus outline is important for keyboard navigation and general accessibility. I hope it isn't overridden often. :(

>  it seems that if you managed to get that outline visible using a mouse, there's something off

The reason it happens with the mouse interaction is because the button still has focus, so it gets the outline. Its not a problem, and its intentional.

<img width="317" alt="screen shot 2018-01-25 at 11 49 16 am" src="https://user-images.githubusercontent.com/191598/35400805-cc70ee96-01c5-11e8-97de-741f2817fcd9.png">

Zooming in, you can see the outline is a little wonky. So I suggest cleaning it up by removing the extra single-pixel border on the left, and rounding all four corners. You'd end up with something like this:

<img width="233" alt="screen shot 2018-01-25 at 11 51 14 am" src="https://user-images.githubusercontent.com/191598/35400911-18e8b966-01c6-11e8-8aec-17e889c9538c.png">

**Comment by andfinally** [comment]
Thanks for explaining @simison and @shaunandrews, that's very useful to know. `accessible-focus` is a brilliant idea. 

It's just dawned on me the outline isn't a user agent style but a `box-shadow` the Calypso CSS adds to buttons in a button group when they have focus. When the `accessible-focus` class is present the style is applied to buttons generally. I won't mess with it.

Your alternative idea for the button styles in my [revisions navigation PR](https://github.com/Automattic/wp-calypso/pull/21712) improves the button shape in a different way Shaun - I reckon I'll do it as you suggest there.
